### PR TITLE
Update `on-slot-render` to TypeScript

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Update `on-slot-render` to TypeScript

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+const recordFirstAdRendered = once(() => {
+	markTime('Commercial: First Ad Rendered');
+});
+
+const reportEmptyResponse = (adSlotId: string, event: SlotRenderEndedEvent) => {
+	// This empty slot could be caused by a targeting problem,
+	// let's report these and diagnose the problem in sentry.
+	// Keep the sample rate low, otherwise we'll get rate-limited (report-error will also sample down)
+	if (Math.random() < 0.0001) {
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts:21]: For your consideration:

```suggestion
	if (Math.random() < 1 / 10_000) {
```

It has a zero-perf cost, according to https://jsbench.me/x1ku71w7za/1

File: static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import mediator from '../../../../lib/mediator';
+import reportError from '../../../../lib/report-error';
+import { markTime } from '../../../../lib/user-timing';
+import { dfpEnv } from './dfp-env';
+import { emptyAdvert } from './empty-advert';
+import { getAdvertById } from './get-advert-by-id';
+import { renderAdvert } from './render-advert';
+
+const recordFirstAdRendered = once(() => {
+	markTime('Commercial: First Ad Rendered');
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts:14]: Isn’t this a duplicate of what we do with the `EventTimer`?

File: static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		// Set refresh field based on the outcome of the slot render.
+		const sizeString = advert.size?.toString();
+		const isNotFluid = sizeString !== '0,0';
+		const isOutstream = outstreamSizes.includes(sizeString as AdSizeString);
+		const isNonRefreshableLineItem =
+			event.lineItemId &&
+			(config.get(
+				'page.dfpNonRefreshableLineItemIds',
+				[],
+			) as string[]).includes(String(event.lineItemId));
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts:90]: Could we please use this as an opportunity to update the config type? 🥺

**Review by mxdvl — APPROVED** [review]
Some small comments, but no blockers. 

Thanks for this!

File: static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts (hunk 2 above)
Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts:14]: Good spot.

We seem to use a mix of `markTime` and `EventTImer` across the commercial codebase.

So I think consolidating this is outside of this PR but I will raise a ticket for it.

**Pull request merged**